src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/DampedHarmonic.cpp

```diff
@@ -0,0 +1,962 @@
+              dt_spatial_metric,                                              \
+          const tnsr::aa<DTYPE(data), DIM(data), FRAME(data)>& pi,            \
+          const tnsr::iaa<DTYPE(data), DIM(data), FRAME(data)>& phi,          \
+          const double g_exponent, const int exponent) noexcept;              \
+  template tnsr::a<DTYPE(data), DIM(data), FRAME(data)> GeneralizedHarmonic:: \
+      DampedHarmonicGauge_detail::spacetime_deriv_of_power_log_factor(        \
+          const Scalar<DTYPE(data)>& lapse,                                   \
+          const tnsr::I<DTYPE(data), DIM(data), FRAME(data)>& shift,          \
+          const tnsr::A<DTYPE(data), DIM(data), FRAME(data)>&                 \
+              spacetime_unit_normal,                                          \
+          const tnsr::II<DTYPE(data), DIM(data), FRAME(data)>&                \
+              inverse_spatial_metric,                                         \
+          const Scalar<DTYPE(data)>& sqrt_det_spatial_metric,                 \
+          const tnsr::ii<DTYPE(data), DIM(data), FRAME(data)>&                \
+              dt_spatial_metric,                                              \
+          const tnsr::aa<DTYPE(data), DIM(data), FRAME(data)>& pi,            \
+          const tnsr::iaa<DTYPE(data), DIM(data), FRAME(data)>& phi,          \
+          const double g_exponent, const int exponent) noexcept;
+
+#define INSTANTIATE_SCALAR_FUNC(_, data)                                     \
+  template void GeneralizedHarmonic::DampedHarmonicGauge_detail::log_factor( \
+      const gsl::not_null<Scalar<DTYPE_SCAL(data)>*> logfac,                 \
+      const Scalar<DTYPE_SCAL(data)>& lapse,                                 \
+      const Scalar<DTYPE_SCAL(data)>& sqrt_det_spatial_metric,               \
+      const double exponent) noexcept;                                       \
+  template Scalar<DTYPE_SCAL(data)>                                          \
+  GeneralizedHarmonic::DampedHarmonicGauge_detail::log_factor(               \
+      const Scalar<DTYPE_SCAL(data)>& lapse,                                 \
+      const Scalar<DTYPE_SCAL(data)>& sqrt_det_spatial_metric,               \
+      const double exponent) noexcept;
+
+#define INSTANTIATE_COMPUTE_ITEM(_, data)                                    \
+  template struct GeneralizedHarmonic::DampedHarmonicHCompute<DIM(data),     \
+                                                              FRAME(data)>;  \
+  template struct GeneralizedHarmonic::SpacetimeDerivDampedHarmonicHCompute< \
+      DIM(data), FRAME(data)>;
+
+GENERATE_INSTANTIATIONS(INSTANTIATE, (1, 2, 3), (double, DataVector),
+                        (Frame::Grid, Frame::Inertial))
+
+GENERATE_INSTANTIATIONS(INSTANTIATE_SCALAR_FUNC, (double, DataVector))
+
+GENERATE_INSTANTIATIONS(INSTANTIATE_COMPUTE_ITEM, (1, 2, 3), (DataVector),
+                        (Frame::Grid, Frame::Inertial))
```

Comment:
This is an open question. While the gauge source was derived in `Inetial` coordinates, the possibility of using grid coordinates has been raised in literature. See, for e.g., the end of Sec 2.2 of https://arxiv.org/pdf/1405.3693.pdf . If the combination of Eq.(21-22) are to be explored in future, then `damped_harmonic_h()`  can be used to get the RHS of Eq.22, using grid frame quantities, and subsequently transformed to inertial coordinates via Eq.21. 

However, till that happens, I can agree that instantiating for `Frame::Grid` is not productive. I have removed it. I've not removed the iteration of `GENERATE_INSTANTIATIONS` over frames though, so that `Frame::Grid` can trivially be added back if/when desired.

Reply:
I like what you've done and agree that what the "best" gauge condition is remains on open question :) Thanks!